frontend/src/main/java/com/google/tripmeout/frontend/servlet/PlaceIndividualServlet.java

```diff
@@ -0,0 +1,116 @@
+package com.google.tripmeout.frontend.servlet;
+
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.error.TripMeOutException;
+import com.google.tripmeout.frontend.places.PlaceService;
+import com.google.tripmeout.frontend.storage.PlaceVisitStorage;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class PlaceIndividualServlet extends HttpServlet {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private static final Pattern URI_NAME_PATTERN =
+      Pattern.compile(".*/trips/([^/]+)/placeVisits/([^/]+)");
+
+  private final PlaceVisitStorage placeStorage;
+  private final Gson gson;
+  private final PlaceService placeService;
+
+  public PlaceIndividualServlet(
+      PlaceVisitStorage placeStorage, Gson gson, PlaceService placeService) {
+    this.placeStorage = placeStorage;
+    this.gson = gson;
+    this.placeService = placeService;
+  }
+
+  @Override
+  public void doPut(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    PrintWriter writer = response.getWriter();
```

Comment:
nit: use a with resources instead of finally...
```java
try (PrintWriter writer = response.getWriter()) {
  // use writer...
}
// writer is automatically closed when we exit the try block scope.
```